### ISSUE TITLE
set some user preferences, custom use case

### ISSUE THREAD

**Issue by ghost** [ShansanChu/insight_DS#3]
Hi @ShansanChu ,

Hope you are all well !

I wanted to ask you a couple of questions as I am developing a website called https://paper2code. 
It is a search engine for research papers and their related source codes.

So, one feature, I wanted to install is really similar to your repository, https://github.com/ShansanChu/insight_DS.

I d like help researcher to browse arxiv/aclweb,etc... papers by using their abstract's text and calculating their similarity. But I would like to add a collaborative layer where if they favorite a paper it allows to improve the ranking of the text similarity results.

So my questions are:
- Is it possible to add such feature in insight ds ?
- Can you help me by sharing some insights or inputs on how to that ? 
- Can I use your project and credit you in the credits page ?

Please find below some articles defining mostly what I d like to do for paper2code.

Refs:
- https://blog.machinebox.io/introducing-suggestionbox-personalization-and-recommendation-machine-learning-in-a-docker-b9d69f937716
- https://medium.com/unstructured-data-service/how-xiaomi-browser-uses-milvus-to-build-its-news-recommender-system-62fb27ba42e4

Thanks in advance for any reply or input on these questions.

Cheers,
X

**Comment by ShansanChu** [comment]
Thanks for your interest in my project. If you are using abstract’s text, I
guess it’s not quite longer. You don’t need to do summarization. One
possible method is to directly use the sentences embedding as feature
extraction. Sure, it’s possible to do personalized ranking by adding more
layers or utilizing ranking models with sentence embedding as input.

As for my project, I don’t have labels for training. What I do with my
project is to retrieve similar articles based on similarity of sentences.
But I don’t have the ranking part. I won’t suggest you to add more layers
based on my project. For your case, you can use sentence embedding of
abstract sentences as feature extraction (obtain embedding input (item’s
character). And then apply ranking models to train your own ranking model.

Cheers,
Shansan

在 2020年7月25日，下午1:22，x0rzkov <notifications@github.com> 写道：

﻿

Hi @ShansanChu <https://github.com/ShansanChu> ,

Hope you are all well !

I wanted to ask you a couple of questions as I am developing a website
called https://paper2code.
It is a search engine for research papers and their related source codes.

So, one feature, I wanted to install is really similar to your repository,
https://github.com/ShansanChu/insight_DS.

I d like help researcher to browse arxiv/aclweb,etc... papers by using
their abstract's text and calculating their similarity. But I would like to
add a collaborative layer where if they favorite a paper it allows to
improve the ranking of the text similarity results.

So my questions are:

   - Is it possible to add such feature in insight ds ?
   - Can you help me by sharing some insights or inputs on how to that ?
   - Can I use your project and credit you in the credits page ?

Please find below some articles defining mostly what I d like to do for
paper2code.

Refs:

   - https://blog.machinebox.io/introducing-suggestionbox-
   personalization-and-recommendation-machine-learning-in-a-docker-
   b9d69f937716
   <https://blog.machinebox.io/introducing-suggestionbox-personalization-and-recommendation-machine-learning-in-a-docker-b9d69f937716>
   - https://medium.com/unstructured-data-service/how-
   xiaomi-browser-uses-milvus-to-build-its-news-recommender-
   system-62fb27ba42e4
   <https://medium.com/unstructured-data-service/how-xiaomi-browser-uses-milvus-to-build-its-news-recommender-system-62fb27ba42e4>

Thanks in advance for any reply or input on these questions.

Cheers,
X

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ShansanChu/insight_DS/issues/3>, or unsubscribe

**Comment by ghost** [comment]
Hi,

Thanks for the reply :-)

In fact, I have hybrid documents, like READMEs, from the code related, that I want to summarize with bert-extractive-summarizer, and the famous abstracts. So I do no know how to flag to skip this step of summarizing during the training process.

Can you help me/us to make a generic solution of insight_DS ? It would be really awesome.

I mean that I am much more a gopher than a pythonista and I am little bit lost with the order to execute the training scripts. For the rest, I mean the server and the frontend stuff, I can manage it with no worries.

If you have 5-10 spare minutes to do it, you would be an awesome guy. What I can offer, it is do dockerize the final stuff in a fork and make a PR.

Cheers,
X

**Comment by ghost** [comment]
@ShansanChu any chance for the generic solution ? please :-)

**Comment by ShansanChu** [comment]
Sorry I have busy schedules these couple weeks. I’ll get back to you when
I’m available.

**Comment by ghost** [comment]
Thanks mate

**Comment by ShansanChu** [comment]
Hi,

Sorry to get back to you later. Returning to the training case, if you want
to skip the summarization step with flag, I would suggest you to use the
summarization as part of your step in data preprocessing rather than adding
this part to the training model. If you want to train the model as part of
the customized ranking, I'm not sure if this works as a end-to-end model.
For the summarization part, to train the model we need to calculate
derivatives during the backpropagation, maybe it would be complicated to
train end-to-end model. To train the summarization part, we need the
labeled dataset.

Best,
Shan

**Comment by ghost** [comment]
Hi,

Thanks for the reply.

I have a dataset of 290k abstracts and here is the link to download it.
http://paper2code.com/public/suggest_dump.txt.tar.gz

Do you think we can give a try to this one ? that would be awesome

Cheers,
X
